modules/lang-painless/src/main/java/org/elasticsearch/painless/node/ERegex.java

```diff
@@ -74,30 +81,90 @@ void analyze(ScriptRoot scriptRoot, Scope scope) {
                     new IllegalArgumentException("Error compiling regex: " + e.getDescription()));
         }
 
-        String name = scriptRoot.getNextSyntheticName("regex");
-        scriptRoot.getClassNode().addField(
-                new SField(location, Modifier.FINAL | Modifier.STATIC | Modifier.PRIVATE, name, Pattern.class));
-        constant = new Constant(location, MethodWriter.getType(Pattern.class), name, this::initializeConstant);
+        name = scriptRoot.getNextSyntheticName("regex");
         actual = Pattern.class;
     }
 
     @Override
-    RegexNode write(ClassNode classNode) {
-        RegexNode regexNode = new RegexNode();
-        regexNode.setLocation(location);
+    MemberFieldLoadNode write(ClassNode classNode) {
+        FieldNode fieldNode = new FieldNode();
+        fieldNode.setLocation(location);
+        fieldNode.setModifiers(Modifier.FINAL | Modifier.STATIC | Modifier.PRIVATE);
+        fieldNode.setFieldType(Pattern.class);
+        fieldNode.setName(name);
 
-        regexNode.setExpressionType(actual);
-        regexNode.setFlags(flags);
-        regexNode.setPattern(pattern);
-        regexNode.setConstant(constant);
+        classNode.addFieldNode(fieldNode);
 
-        return regexNode;
-    }
+        try {
+            StatementExpressionNode statementExpressionNode = new StatementExpressionNode();
+            statementExpressionNode.setLocation(location);
+            statementExpressionNode.setMethodEscape(true);
+            statementExpressionNode.setNoop(true);
+
+            BlockNode blockNode = classNode.getClinitNode().getBlockNode();
+            blockNode.addStatementNode(statementExpressionNode);
+
+            MemberFieldStoreNode memberFieldStoreNode = new MemberFieldStoreNode();
+            memberFieldStoreNode.setLocation(location);
+            memberFieldStoreNode.setExpressionType(Pattern.class);
+            memberFieldStoreNode.setName(name);
+            memberFieldStoreNode.setStatic(true);
+
+            statementExpressionNode.setExpressionNode(memberFieldStoreNode);
+
+            CallNode callNode = new CallNode();
+            callNode.setLocation(location);
+            callNode.setExpressionType(Pattern.class);
+
+            memberFieldStoreNode.setChildNode(callNode);
+
+            StaticNode staticNode = new StaticNode();
+            staticNode.setLocation(location);
+            staticNode.setExpressionType(Pattern.class);
+
+            callNode.setLeftNode(staticNode);
+
+            CallSubNode callSubNode = new CallSubNode();
+            callSubNode.setLocation(location);
+            callSubNode.setExpressionType(Pattern.class);
+            callSubNode.setBox(Pattern.class);
+            callSubNode.setMethod(new PainlessMethod(
+                    Pattern.class.getMethod("compile", String.class, int.class),
```

Comment:
Where is this being written to `clinit`?

Reply:
Oh perhaps in `classNode.addFieldNode(fieldNode);`?

Reply:
`classNode.addFieldNode` creates a declaration for a member field.

These lines actually add the regex constant generation and assignment to the clinit method:

```
BlockNode blockNode = classNode.getClinitNode().getBlockNode();
blockNode.addStatementNode(statementExpressionNode);
```